pkg/infra/httpclient/httpclientprovider/datasource_metrics_middleware.go

```diff
@@ -27,7 +28,7 @@ var (
 			Name:      "datasource_request_duration_seconds",
 			Help:      "histogram of durations of outgoing data source requests sent from Grafana",
 			Buckets:   []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10, 25, 50, 100},
-		}, []string{"datasource", "datasource_type", "code", "method"},
+		}, []string{"datasource", "datasource_type", "code", "method", "secureSocksDSProxyEnabled"},
```

Comment:
To follow existing label convention using snake case styling

```suggestion
		}, []string{"datasource", "datasource_type", "code", "method", "secure_socks_enabled"},
```

Reply:
Renamed to `secure_socks_ds_proxy_enabled`